### ISSUE TITLE
Auto Sort is slow on huge boomark set

### ISSUE THREAD

**Issue by Morikko** [eric-bixby/auto-sort-bookmarks-webext#108]
If the auto sort function is on then any change on the bookmark is getting really slow.

Based [on the code](https://github.com/eric-bixby/auto-sort-bookmarks-webext/blob/799f78fe96a68c6bea3e17b4e9c607025708eb8c/src/background/auto-sort-bookmarks.js#L1156), I suspect that it sorts all the bookmark all the time and it takes seconds for my set.

I do not know the API by heart but I suspect that it should be possible to know which bookmark changed (CUD) and to sort only the changed bookmark folder level.

Note: I would even think that a delete should not require a sort as the elements were already sorted.

**Comment by eric-bixby** [comment]
Originally, that's how Auto-Sort Bookmarks (ASB) worked.  The problem was there are edge cases where bookmarks would not get sorted, so logic was added but this didn't always work and caused other problems (like corruption).  For the sake of simplicity and stability, it does a delayed sort (wait for no activity, then sort).  I had planned to restore the delta sort, but haven't gotten around to it yet.

**Comment by Gitoffthelawn** [comment]
Of course, continuing to prevent corruption needs to be of the highest priority!
